index.js

```diff
@@ -169,4 +169,35 @@ const isEmptyMapOrSet = x => (is.map(x) || is.set(x)) && x.size === 0;
 
 is.empty = x => !x || isEmptyStringOrArray(x) || isEmptyObject(x) || isEmptyMapOrSet(x);
 
+const isAnyOrAll = (all, predicate, values) => {
+	// `values` is the calling function's "arguments object".
+	// We have to do it this way to keep node v4 support.
+	// So here we convert it to an array and slice off the first item.
+	values = Array.prototype.slice.call(values, 1);
+
+	if (is.function(predicate) === false) {
+		throw new TypeError(`Invalid predicate: ${util.inspect(predicate)}`);
+	}
+
+	if (values.length === 0) {
+		throw new TypeError(`Invalid number of values: ${util.inspect(values.length)}`);
+	}
+
+	if (all) {
+		return values.every(predicate);
+	}
+
+	return values.some(predicate);
+};
+
+// We have to use anonymous functions for the any() and all() methods
+// to get the arguments since we can't use rest parameters in node v4.
+is.any = function (predicate) {
```

Comment:
`is.any = predicate => isAnyOrAll(...)`

Reply:
Have to use an anonymous function in order to get the correct arguments object. For some reason the arguments object is weird with arrow functions.

Reply:
God I hate old js.

Reply:
https://stackoverflow.com/questions/30935336/official-information-on-arguments-in-es6-arrow-functions

Arrow functions dont have arguments.

Reply:
Just for the sake of debugging, name the anonymous functions (isAll and isAny) so that they'll be named if they appear in a stack trace.

After that, I'll approve

Reply:
Also, we now have `is.empty` for `length === 0`

Reply:
> Just for the sake of debugging, name the anonymous functions (isAll and isAny) so that they'll be named if they appear in a stack trace.

That is not necessary. JS engines are able to infer the name.

Reply:
> Also, we now have is.empty for length === 0

I think it's overdoing it using `is.empty` for that.